src/libos/src/net/socket/unix_socket/stream/socket.rs

```diff
@@ -0,0 +1,190 @@
+use super::*;
+use alloc::sync::Arc;
+use std::fmt;
+use std::sync::atomic::{AtomicBool, Ordering};
+
+/// SOCK_STREAM Unix socket
+pub struct StreamUnix {
+    path: RwLock<UnixAddr>,
+    end: SgxMutex<Option<Arc<UnixEnd>>>, // Set after connection
+    is_blocking: AtomicBool,
+}
+
+impl StreamUnix {
+    pub fn new(flags: FileFlags) -> Result<Self> {
+        Ok(Self {
+            path: RwLock::new(UnixAddr::UnNamed),
+            end: SgxMutex::new(None),
+            is_blocking: AtomicBool::new(!flags.contains(FileFlags::SOCK_NONBLOCK)),
+        })
+    }
+
+    pub fn path(&self) -> UnixAddr {
```

Comment:
I think the signature of the method would be better to be like the following:
```
pub fn addr(&self) -> &UnixAddr {
```
which saves a copy and makes the name more accurate.

Reply:
The value is inside a rwlock. It seems hard to borrow a value from inside rwlock.